feat: 공고 목록 페이지 UI 및 기능 구현 (#34)

KoCeleste opened the pull request:
<!-- PR 제목 규칙:
[type]: 작업 요약 (#이슈번호)
예시: feat: 로그인 페이지 UI 구현 (#12)
-->

## #️⃣ 연관된 이슈

- Resolves #34 

## 📑 구현 사항

- 검색 및 필터링, 정렬 기능
- Load More 페이지네이션
- 로그인 상태별 UI(비로그인/로그인)
- 리베이스 과정 중 발생한 충돌 해결 및 파일 구조 정리
- Input 컴포넌트를 복구하고 타입 오류 수정
- tsconfig 설정 정리하여 JSX/타입 오류 해결
- 비회원 유저에게 CardSkeleton 자동 표시되도록 기능 추가
- 불필요한 파일 및 중복 코드 제거하여 전체 구조 안정화

## 🌀 어려웠던 점 / 고민했던 부분

- PC에서는 자연스러운데 모바일에서 레이아웃이 깨지는 문제
- 맞춤 추천 카드의 북마크/조회수 아이콘이 썸네일을 가리는 이슈
: 아이콘을 제목 옆으로 이동하여 PC/모바일 모두 깔끔하게 표시

- 스크롤 이벤트를 useState로 처리하려다 메모리 누수 발생 가능성 발견
: useEffect로 변경하여 cleanup 함수로 이벤트 리스너 제거
 

## 📸 구현 이미지

![로그인1](https://github.com/user-attachments/assets/b82c6d03-830e-42a5-b0d8-cd00c9f747ba)
![로그인2](https://github.com/user-attachments/assets/4f4a7d7b-94d1-48a5-989a-221af0551434)
![로그인3](https://github.com/user-attachments/assets/bf8bdaeb-4d30-4473-8ee5-bfed2c5335ce)
![비로그인1](https://github.com/user-attachments/assets/103fdf61-2ae0-468d-9a07-9614795af8d5)
![비로그인2](https://github.com/user-attachments/assets/8ad64f5b-a060-43c7-ba83-6eee34405182)
![검색](https://github.com/user-attachments/assets/a18755ac-91fd-4041-bcc7-9562ab16618d)
![필터링](https://github.com/user-attachments/assets/3a15ce5a-04bb-40ae-99fc-1dcf54320af4)
![필터링2](https://github.com/user-attachments/assets/517942d4-ffa2-48e0-bf59-f5ffb272a46f)


## ❇️ 코드 설명
 
## 💬 리뷰 요청사항

> 리뷰어에게 피드백을 받고 싶은 지점이 있다면 작성해주세요.

KoCeleste committed:
refactor: Input 컴포넌트 경로 이동

- src/components/ui → src/components/common
- 샘플 파일 삭제 및 input 컴포넌트 경로 이동

KoCeleste committed:
feat: 스터디 공고 목록 페이지 구현

- 공고 검색, 필터링, 정렬 기능
- 로그인/비로그인 UI 분기 (화면 확인용)
- 추천 공고 인기도 기반 알고리즘
- 반응형 디자인 및 페이지네이션
- Mock 데이터(임의로 15개) 추가

KoCeleste committed:
fix: Input 컴포넌트 복구 및 타입(className) 오류 수정
- import 경로 수정
- 빌드 실패 원인 해결

agrade1 commented:
사용자 맞춤 공고 스켈레톤 화면 예은님이 구현하신거로 사용이 안되어있는거같은대 한번 확인 부탁드려요!
@KoCeleste

yeeun-kor commented on src/components/common/input.tsx:
헉스 ...common에도 input이 있고, components의 input도 있고,,,, 조금 복잡해서 어떤걸 사용해야하는지 헷갈립니다!!!

yeeun-kor commented on src/routes/AppRoutes.tsx:
```diff
@@ -1,18 +1,45 @@
       <Route element={<Layout />}>
-        {/* 이곳에 라우팅 추가하기 */}
-        <Route index element={<Main />} />
-        <Route path="/courses" element={<Courses></Courses>}></Route>
```
엇 이건 제가 쓰는 페이지인데, (수강목록 페이지) 
지우신건가요 1?!?!

yeeun-kor commented on src/pages/postings/RecruitmentListPage.tsx:
```diff
@@ -0,0 +1,472 @@
+                공고를 추천해드립니다
+              </p>
+              <div className="mb-6 flex flex-col justify-center gap-4 sm:flex-row">
+                <button className="flex items-center justify-center gap-2 rounded-lg bg-yellow-400 px-8 py-3 text-base font-medium text-white transition-colors hover:bg-yellow-500">
```
연우님 이 부분은 제가 만들어두었어요! 
여기 pr참고해주세요 
https://github.com/OZ-Coding-School/oz_externship_fe_04_team3/pull/46

비회원 추천 섹션 사용하기 입니다!

KoCeleste committed:
fix: rebase 이후 빌드/타입 오류 해결 및 구조 정리
- input 파일 위치 이동 및 icon 코드 삭제
- AppRouter 파일 예은님 코드 복구
- 비회원 유저 카드 스켈레톤 적용
- tsconfig 설정 정리로 JSX/경로 관련 빌드 오류 해결

agrade1 commented:
필터링 로직 msw 핸들러로 분리 필요
카드 컴포넌트 posting/manage/manageCard 사용 ( click 이벤트 props 처리 필요 )
listPage 로직 분리 필요.

soohyun-bae commented on src/components/input.tsx:
```diff
@@ -1,5 +1,3 @@
-import React from 'react'
-
 type InputProps = Omit<
   React.InputHTMLAttributes<HTMLInputElement>,
   'prefix' | 'suffix'
```
Omit 사용한 이유가 있을까용??

soohyun-bae commented on src/components/input.tsx:
```diff
@@ -24,7 +22,7 @@ export function Input({
   prefix,
   suffix,
-  icon,
+  icon: _icon,
```
요고는 사용을 안해서 언더스코어 넣으신걸까용?

soohyun-bae commented on src/mocks/recruitmentData.ts:
이 목데이터는 어떤 api 호출에 대한 데이터로 만들어둔걸까용???

soohyun-bae reviewed (dismissed):
수고하셨숩니다아아

KoCeleste committed:
feat: 공고 카드 UI 리팩토링 및 추천 공고 섹션 개선

- 추천 공고 이미지 영역 전체 채우기 적용
- 공고 카드 구조 통일 및 중복 코드 정리
- RecruitmentListPage 검색/정렬/카테고리 로직 정비
- useRecruitments 훅 구조 정리와 추천 데이터 계산 개선

KoCeleste committed:
refactor: MSW 핸들러로 공고 필터링 로직 분리

- mocks/handlers/recruitments 핸들러 생성
- mocks/handlers.ts에 recruitment Handlers 등록
- api/axios.ts에getRecruitments 함수 추가
- useRecruitments 훅 API 호출방식으로 변경
- RecruitmentListPage 로딩상태 처리 추가

KoCeleste committed:
fix: handlers.ts 충돌해결

- recruitments, myRecruitments 핸들러 분리와 누락된 부분 추가
